### PR TITLE
DAOS-3085 control: Fix go test runner environment

### PR DESCRIPTION
**Pull request by mjmac**
Missed this in local/CI testing because the variables were
set externally. In the case where build environment variables
need to be set, provide default empty values to avoid triggering
unset variable errors. Also adds hwloc path to environment.

### CODE REVIEW

File: src/control/run_go_tests.sh [hunk 1]
```diff
@@ -20,15 +20,15 @@ function find_build_source()
 
 function check_environment()
 {
-	if [ -z "$LD_LIBRARY_PATH" ]; then
+	if [ -z "${LD_LIBRARY_PATH:-}" ]; then
```
Comment (brianjmurrell) [inline src/control/run_go_tests.sh]: Am I missing the subtlety of the empty `${...:-}` syntax here?

```bash
$ [ -z "$foo" ] && echo empty
empty
$ [ -z "${foo:-}" ] && echo empty
empty
```

File: src/control/run_go_tests.sh [hunk 2]
```diff
@@ -44,10 +44,16 @@ function setup_environment()
...
 	source "${build_source}"
 
-	LD_LIBRARY_PATH="${SL_PREFIX}/lib:${SL_SPDK_PREFIX}/lib:${LD_LIBRARY_PATH}"
-	export LD_LIBRARY_PATH
-	export CGO_LDFLAGS="-L${SL_SPDK_PREFIX}/lib -L${SL_PREFIX}/lib"
-	export CGO_CFLAGS="-I${SL_SPDK_PREFIX}/include"
+	# ugh, appease the linter...
+	LD_LIBRARY_PATH=${SL_PREFIX}/lib
+	LD_LIBRARY_PATH+=":${SL_SPDK_PREFIX}/lib"
+	LD_LIBRARY_PATH+=":${SL_HWLOC_PREFIX}/lib"
```
Comment (brianjmurrell) [inline src/control/run_go_tests.sh]: It's your intention to overwrite and not just extend `LD_LIBRARY_PATH` here?

File: src/control/run_go_tests.sh (hunk 1 above)
Comment (mjmac) [reply src/control/run_go_tests.sh]: The first one will fail with `set -u` (which is enabled now).

File: src/control/run_go_tests.sh (hunk 2 above)
Comment (mjmac) [reply src/control/run_go_tests.sh]: For tests, I'd prefer to, yes. Unless someone can think of a good idea not to. My thought was that it ensures that the tests either happen with an environment completely supplied from the outside, or else the test runner handles it completely. I'd kind of like to disallow external environment altogether but figured that might make life harder for local development.

**Pull request merged**